pkgs/servers/misc/navidrome/default.nix

```diff
@@ -1,50 +1,61 @@
-{ lib, stdenv, fetchurl, ffmpeg, ffmpegSupport ? true, makeWrapper, nixosTests }:
+{ callPackage, buildGoModule, fetchFromGitHub, lib, pkg-config, stdenv, ffmpeg, taglib, zlib, ffmpegSupport ? true, makeWrapper, nixosTests }:
 
 with lib;
 
-stdenv.mkDerivation rec {
-  pname = "navidrome";
+let
+
   version = "0.47.5";
 
+  src = fetchFromGitHub {
+    owner = "navidrome";
+    repo = "navidrome";
+    rev = "v${version}";
+    hash = "sha256-gTvJI+brdEpdpbEcdQycqw15seI+k5dMDVrjY3v6i14=";
+  };
+
+  ui = callPackage ./ui {
+    inherit src version;
+  };
 
-  src = fetchurl (if stdenv.hostPlatform.system == "x86_64-linux"
-  then {
-    url = "https://github.com/deluan/navidrome/releases/download/v${version}/navidrome_${version}_Linux_x86_64.tar.gz";
-    sha256 = "sha256-AkSjtln53HDdIcQgnA8Wj010RXnOlOsFm2wfVgbvwtc=";
-  }
-  else {
-    url = "https://github.com/deluan/navidrome/releases/download/v${version}/navidrome_${version}_Linux_arm64.tar.gz";
-    sha256 = "sha256-+VBRiV2zKa6PwamWj/jmE4iuoohAD6oeGnlFi4/01HM=";
-  });
+in
 
-  nativeBuildInputs = [ makeWrapper ];
+buildGoModule {
 
-  unpackPhase = ''
-    tar xvf $src navidrome
-  '';
+  pname = "navidrome";
+
+  inherit src version;
+
+  vendorSha256 = "sha256-xMAxGbq2VSXkF9R9hxB9EEk2CnqsRxg2Nmt7zyXohJI=";
 
-  installPhase = ''
-    runHook preInstall
+  nativeBuildInputs = [ makeWrapper pkg-config ];
 
-     mkdir -p $out/bin
-     cp navidrome $out/bin
+  buildInputs = [ taglib zlib ];
 
-    runHook postInstall
+  CGO_CFLAGS = [ "-Wno-return-local-addr" ];
+
+  prePatch = ''
+    cp -r ${ui}/* ui/build
   '';
 
   postFixup = optionalString ffmpegSupport ''
     wrapProgram $out/bin/navidrome \
       --prefix PATH : ${makeBinPath [ ffmpeg ]}
   '';
 
-  passthru.tests.navidrome = nixosTests.navidrome;
+  doCheck = false;
+
+  passthru = {
+    inherit ui;
+    tests.navidrome = nixosTests.navidrome;
+    updateScript = callPackage ./update.nix {};
+  };
 
   meta = {
     description = "Navidrome Music Server and Streamer compatible with Subsonic/Airsonic";
     homepage = "https://www.navidrome.org/";
-    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
     license = licenses.gpl3Only;
-    platforms = [ "x86_64-linux" "aarch64-linux" ];
-    maintainers = with maintainers; [ aciceri ];
+    sourceProvenance = with lib.sourceTypes; [ fromSource ];
```

Comment:
The contents of `lib` are already in scope due to the `with` in L3.  I'd prefer if we can go without the `with lib`, or at least not use it on the whole derivation, because as you noticed earlier `lib.version` was not what you wanted.

Reply:
Removed the explicit call to lib